convert2rhel/unit_tests/actions/system_checks/readonly_mounts_test.py

```diff
@@ -17,109 +17,123 @@
 
 __metaclass__ = type
 
-import unittest
-
 import six
 
-from convert2rhel import actions, unit_tests
+from convert2rhel import unit_tests
 from convert2rhel.actions.system_checks import readonly_mounts
-from convert2rhel.unit_tests import GetFileContentMocked, GetLoggerMocked
 
 
 six.add_move(six.MovedModule("mock", "mock", "unittest.mock"))
+import pytest
+
 from six.moves import mock
 
 
-class TestReadOnlyMountsChecks(unittest.TestCase):
-    def setUp(self):
-        self.readonly_mounts_action_mnt = readonly_mounts.ReadonlyMountMnt()
-        self.readonly_mounts_action_sys = readonly_mounts.ReadonlyMountSys()
+@pytest.fixture
+def readonly_mounts_mnt():
+    return readonly_mounts.ReadonlyMountMnt()
+
 
-    @unit_tests.mock(readonly_mounts, "logger", GetLoggerMocked())
-    @unit_tests.mock(
+@pytest.fixture
+def readonly_mounts_sys():
+    return readonly_mounts.ReadonlyMountSys()
+
+
+def test_mounted_mnt_is_readwrite(readonly_mounts_mnt, caplog, monkeypatch):
+    monkeypatch.setattr(
         readonly_mounts,
         "get_file_content",
-        GetFileContentMocked(
-            data=[
+        mock.Mock(
+            return_value=[
                 "sysfs /sys sysfs ro,seclabel,nosuid,nodev,noexec,relatime 0 0",
                 "mnt /mnt sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0",
                 "cgroup /sys/fs/cgroup/cpuset cgroup rw,seclabel,nosuid,nodev,noexec,relatime,cpuset 0 0",
             ]
         ),
     )
-    def test_mounted_mnt_is_readwrite(self):
-        self.readonly_mounts_action_mnt.run()
-        print(readonly_mounts.logger.debug_msgs)
-        self.assertEqual(len(readonly_mounts.logger.debug_msgs), 1)
-        self.assertIn("/mnt mount point is not read-only.", readonly_mounts.logger.debug_msgs)
-
-    @unit_tests.mock(readonly_mounts, "logger", GetLoggerMocked())
-    @unit_tests.mock(
+    readonly_mounts_mnt.run()
+
+    unit_tests.assert_actions_result(
+        readonly_mounts_mnt,
+        level="SUCCESS",
+    )
+    assert "Read-only /mnt mount point not detected." in caplog.text
```

Comment:
I wonder if we can assert something else than debug

Reply:
This is actually an info message :-).

We probably can rely on the `unit_tests.assert_actions_result()` rather than checking for the logging output at all, though.  The result is equivalent to the return value of a function and return values are normally what you would assert in unittests.  Depending on how much information we pass to `assert_action_result` (Just passing level nad id, for instance) it would also fix the annoyance of having to fix these unittests anytime the log message strings are reworded.

Since we're using logging to display information to the end user, though, I am not sure if we want to go that route or not (so it's probably better to talk about it in cabal, make a decision, and then implement it in one or more other PRs).